src/Constraints/Collection.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Tuf\Constraints;
+
+use Symfony\Component\Validator\Constraints\Collection as SymfonyCollection;
+
+/**
+ * Custom constraint that extends Symfony's Collection constraint to add 'excludedFields'.
+ */
+class Collection extends SymfonyCollection
+{
+
+    /**
+     * @var array
+     */
+    public $unsupportedFields = [];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __construct($options = null)
+    {
+        if (isset($options['excludedFields']) && !isset($options['fields'])) {
```

Comment:
Is it okay if $options['fields'] is an empty array, or other falsy value? If not, we might want to use `empty()` instead.

Reply:
I actually removed the overridden constructor completely. I don't think we need this check here. It could be the case where would want to use this Collection class without specifying any `fields` in that case you don't care which fields **are** declared just that some fields should not be.